### DISCUSSION TITLE
Adminer vs Elastisearch

### DISCUSSION THREAD

**Discussion by LionelLaffineur** [adminerevo/adminerevo#92] [Q&A]
Someone with experience connecting Adminer & Elastisearch would like to share about connection information / details ?
Which values to use in the Adminer login page ?
I've never used Elastisearch and can't find much information.

Thanks in advance !

**Comment by peterpp** [comment]
Hello Lionel,
you need to use plugin `AdminerLoginPasswordLess` with defined custom password (in the same way as in sqlite.php):
```php
new AdminerLoginPasswordLess(password_hash("mypassword", PASSWORD_DEFAULT)),
```
After that, you should be able to connect to Elasticsearch with server (e.g localhost:9200) and password.

It's important to say that the current state of Elastic plugin is really not usable. It was developed 5-6 years ago in Elasticsearch 5 era and fixed for Elastic 6 (by me). However, I'm using it since then and I prepared lots of fixes and improvements, even compatibility with Elastic 7:
https://github.com/vrana/adminer/pull/422
https://github.com/vrana/adminer/pull/420

Finally, I made a new driver for Elastic 7:
https://github.com/vrana/adminer/pull/440
because internal architecture of Elastic indices has been significantly changed and it can (and should) be displayed differently.

I'm willing to prepare MR to AdminerEvo, but there are few questions to discuss:
1) Is it a good idea to keep Elastic as an internal driver? In my opinion not. It is not an essential driver (like MySQL or PostgreSQL).
2) Is it necessery to keep support for old versions (Elastic 5 and 6)?
3) Can be the driver for Elastic 7 separated from older versions? It would be really a great pain to keep them together.

**Comment by LionelLaffineur** [comment]
Dear @peterpp, if you have experience with Adminer and Elastisearch, would you like to take care of that driver ?
Even if I would like, I am not able to fix everything alone ... because I cannot know everything about everything and because there is "only" 24 hours in a day ;-)
Normally, for every PR, I make a set of simple tests to validate it does the job.  In this case, I'm not sure I will succeed in having a setup but I will try with your help.

**Reply by peterpp** [reply]
I can take care of it as long as it fits to my needs and knowledge. And what is your opinion about my questions?
